server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -372,4 +461,14 @@ protected String getMasterActionName(DiscoveryNode node) {
         return getClusterManagerActionName(node);
     }
 
+    /**
+     * Override to true if the transport action need NOT be executed always on cluster-manager (example Read-only actions).
+     * The action is executed locally if this method returns true AND
+     * the ClusterState on local node is in-sync with ClusterManager.
+     *
+     * @return - boolean if the action can be run locally
+     */
+    protected boolean canUseLocalNodeClusterState() {
```

Comment:
there is a method localExecute() in ```TransportClusterManagerNodeAction``` .  should we call localExecuteSupportedForAction? Thoughts?

Reply:
I noticed this method but thought of not renaming it similar as the two decisions (based on request vs latest cluster-state) are independent. How about introducing overloaded version of `localExecute(ClusterState)` ?

Reply:
Yes local `ClusterState` is not required currently. Changed to `localExecuteSupportedForAction`.